pymeasure/instruments/etslindgren/m7006_001.py

```diff
@@ -0,0 +1,174 @@
+
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2022 PyMeasure Developers
+
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+
+import logging
+import re
+from datetime import datetime, timezone, timedelta
+from turtle import position
+import numpy as np
+import pandas as pd
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import (
+    strict_discrete_set, strict_discrete_range,
+    truncated_discrete_set, strict_range
+)
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+
+class M7006_001(Instrument):
+    """ Position control card of EMCenter modular system.
+        Allows for controll and feedback of the positioner card.
+        Comunication happes transparently through the EMCenter mainframe.
+        devices share the resource name of the mainframe, and 
+        are identified by the slot number and the device number(if applicable to the modual).
+    """
+
+    def __init__(self,resource_name, slot=1, device='A', **kwargs):
+
+        kwargs.setdefault('write_termination', '\n')
+        super().__init__(
+            resource_name,
+            "test",
+            **kwargs
+        )
+        self._slot = slot
+        self._device = device
+
+  
+    @property
+    def acceleration(self):
+        """ Returns the acceleration of the device, im seconds """
+        return self.adapter.ask(f'{self._slot}lf{self._device}ACC?')
+    
+    @acceleration.setter
+    def acceleration(self, value):
+        """ Sets the acceleration of the device, in seconds. vaule must be between 0.1 and 30.0 """
+        value = strict_range(value, 0.1, 30.0)
+        return self.adapter.ask(f'{self._slot}{self._device}ACC {value}')
```

Comment:
As coding style, we try to use our own [style of properties](https://pymeasure.readthedocs.io/en/latest/dev/adding_instruments.html#advanced-properties).

Besides, you do not have to rely on the adapter directly, but can use self.read/self.write/self.values. These methods can contain the string formatting as well.

Here is an example for the rewritten methods and how to write a property.

```suggestion
    def write(self, command):
        super().write(f"self._slot}{self._device}{command}")

    def values(self, command, **kwargs):
        return super().values(f"self._slot}{self._device}{command}", **kwargs)

    acceleration = Instrument.control(
        "ACC?", "ACC %g",
        """Acceleration of the device in seconds. Settable 0.1 and 30, and gettable.""",
        values=(0.1, 30),
        validator=strict_range,
        )

Reply:
You can enable "check_errors" for the getter and write a check_errors method (I have to look up the correct names tomorrow), which reads the response.

That is the preferred way dealing with setters which return an answer.

Reply:
I ended up not needing errors.